File: src/prod/src/data/txnreplicator/loggingreplicator/PeriodicCheckpointTruncationState.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace ktl;

namespace Data
{
    namespace LoggingReplicator
    {
        namespace PeriodicCheckpointTruncationState
        {
            void WriteToTextWriter(Common::TextWriter & w, Enum const & val)
            {
                switch (val)
                {
                case NotStarted:
                    w << L"NotStarted"; return;
                case Ready:
                    w << L"Ready"; return;
                case CheckpointStarted:
                    w << L"CheckpointStarted"; return;
                case CheckpointCompleted:
                    w << L"CheckpointCompleted"; return;
                case TruncationStarted:
                    w << L"TruncationStarted"; return;
                default:
                    CODING_ASSERT("Unknown TruncationState:{0} in Replicator", val);
                }
            }

            ENUM_STRUCTURED_TRACE(PeriodicCheckpointTruncationState, NotStarted, LastValidEnum)
        }
    }
}

File: src/prod/src/managed/Microsoft.ServiceFabric.Data.Impl/Replicator/PeriodicCheckpointTruncationState.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace Microsoft.ServiceFabric.Replicator
{
    /// <summary>
    /// Various phases for periodic checkpointing + truncation. Provides guarantee that log does not contain items older than configured interval
    /// </summary>
    public enum PeriodicCheckpointTruncationState : int
    {
        /// <summary>
        /// Configured interval has not yet passed
        /// </summary>
        NotStarted = 0,

        /// <summary>
        /// Configured interval has passed. 
        /// Subsequent call to 'ShouldCheckpoint' return true (if no checkpoint in progress), overriding checkpointIntervalBytes configuration.
        /// </summary>
        Ready = 1,

        /// <summary>
        /// Checkpoint initiated
        /// </summary>
        CheckpointStarted = 2,

        /// <summary>
        /// Checkpoint completed. 
        /// Subsequent calls to 'ShouldTruncateHead' return true (if no truncation in progress), overriding truncation threshold and min log size configurations
        /// </summary>
        CheckpointCompleted = 3,

        /// <summary>
        /// Truncation has been initiated
        /// </summary>
        TruncationStarted = 4
    }
}